web/src/routes/auth/pin-prompt/+page.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+  import { goto } from '$app/navigation';
+  import AuthPageLayout from '$lib/components/layouts/AuthPageLayout.svelte';
+  import PinCodeCreateForm from '$lib/components/user-settings-page/PinCodeCreateForm.svelte';
+  import PincodeInput from '$lib/components/user-settings-page/PinCodeInput.svelte';
+  import { AppRoute } from '$lib/constants';
+  import { handleError } from '$lib/utils/handle-error';
+  import { verifyPinCode } from '@immich/sdk';
+  import { Icon } from '@immich/ui';
+  import { mdiLockOpenVariantOutline, mdiLockOutline } from '@mdi/js';
+  import { t } from 'svelte-i18n';
+  import { fade } from 'svelte/transition';
+  import type { PageData } from './$types';
+
+  interface Props {
+    data: PageData;
+  }
+
+  let { data }: Props = $props();
+
+  let isVerified = $state(false);
+  let isBadPinCode = $state(false);
+  let hasPinCode = $derived(data.hasPinCode);
+  let pinCode = $state('');
+
+  const onPinFilled = async (code: string) => {
+    try {
+      await verifyPinCode({
+        pinCodeSetupDto: {
+          pinCode: code,
+        },
+      });
+
+      isVerified = true;
+
+      let redirectPath = `${AppRoute.LOCKED}`;
+      if (data.continuePath) {
+        redirectPath += `/${data.continuePath}`;
```

Comment:
we should redirect to the exact path and it should not be restricted to sub paths of `/locked` imo